boilerplate/openshift/golang-osd-operator/csv-generate/catalog-build.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+usage() { echo "Usage: $0 -o operator_name -i operator_image -c saas-repository-channel [-t]" 1>&2; exit 1; }
+
+while getopts "o:h:c:" option; do
+    case "${option}" in
+        o)
+            operator_name=${OPTARG}
+            ;;
+        c)
+            operator_channel=${OPTARG}
+            ;;
+        h)
+            operator_commit_hash=${OPTARG}
+            ;;
+        *)
+            usage
+    esac
+done
+
+# Checking parameters
+unset csv_missing_param_error
+
+if [ -z "$operator_channel" ] ; then 
+    echo "Missing operator_channel parameter"
+    csv_missing_param_error = true
+fi
+
+if [ -z "$operator_name" ] ; then 
+    echo "Missing operator_name parameter"
+    csv_missing_param_error = true
+fi
+
+if [ -z "$operator_commit_hash" ] ; then 
+    echo "Missing operator_commit_hash parameter"
+    csv_missing_param_error = true
+fi
+
+if [ ! -z "$csv_missing_param_error" ] ; then
+    usage
+fi
```

Comment:
Since you're using this pattern a lot for mandatory params, you could DRY it up with a common function like:

```bash
function check_mandatory_params() {
    local csv_missing_param_error
    local param_name
    local param_val
    for param_name in "$@"; do
        eval param_val=\$$param_name
        if [ -z "$param_val" ]; then
            echo "Missing $param_name parameter"
            csv_missing_param_error=true
        fi
    done
    if [ ! -z "$csv_missing_param_error" ]; then
        usage
    fi
}
```

Then just e.g.

```bash
check_mandatory_params operator_channel operator_name operator_commit_hash
```